v2/jrl-cmakemodules-v2/modules/jrl.cmake

```diff
@@ -0,0 +1,1788 @@
+    jrl_find_package(Python ${ARGN})
+
+    # On Windows, Python_SITELIB returns \. Let's convert it to /.
+    cmake_path(CONVERT ${Python_SITELIB} TO_CMAKE_PATH_LIST Python_SITELIB NORMALIZE)
+
+    message(STATUS "   Python_FOUND             : ${Python_FOUND}")
+    message(STATUS "   Python_EXECUTABLE        : ${Python_EXECUTABLE}")
+    message(STATUS "   Python_VERSION           : ${Python_VERSION}")
+    message(STATUS "   Python_SITELIB           : ${Python_SITELIB}")
+    message(STATUS "   Python_INCLUDE_DIRS      : ${Python_INCLUDE_DIRS}")
+    message(STATUS "   Python_LIBRARIES         : ${Python_LIBRARIES}")
+    message(STATUS "   Python_SOABI             : ${Python_SOABI}")
+    message(STATUS "   Python_NumPy_FOUND       : ${Python_NumPy_FOUND}")
+    message(STATUS "   Python_NumPy_VERSION     : ${Python_NumPy_VERSION}")
+    message(STATUS "   Python_NumPy_INCLUDE_DIRS: ${Python_NumPy_INCLUDE_DIRS}")
+endmacro()
+
+# Shortcut to find the nanobind package
+# Usage: jrl_find_nanobind()
+macro(jrl_find_nanobind)
+    string(REPLACE ";" " " args_pp "${ARGN}")
+    jrl_check_var_defined(Python_EXECUTABLE "Python executable not found (variable Python_EXECUTABLE).
+
+    Please call jrl_find_python(<args>) first, e.g.:
+
+        jrl_find_python(3.8 REQUIRED COMPONENTS Interpreter Development.Module)
+        jrl_find_package(nanobind ${args_pp})
+    "
+    )
+    unset(args_pp)
+
+    # Detect the installed nanobind package and import it into CMake
+    # ref: https://nanobind.readthedocs.io/en/latest/building.html#finding-nanobind
+    execute_process(
+        COMMAND ${Python_EXECUTABLE} -m nanobind --cmake_dir
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+        OUTPUT_VARIABLE nanobind_ROOT
+        ERROR_VARIABLE nanobind_error
+    )
+
+    if(nanobind_error)
+        unset(nanobind_ROOT)
+
+        # On Ubuntu 24.04, nanobind installed via apt is located in /usr/share/nanobind
+        find_path(nanobind_INCLUDE_DIR NAMES nanobind/nanobind.h HINTS /usr/share/nanobind/include)
+        if(nanobind_INCLUDE_DIR)
+            set(nanobind_ROOT ${nanobind_INCLUDE_DIR}/../cmake)
+            cmake_path(CONVERT ${nanobind_ROOT} TO_CMAKE_PATH_LIST nanobind_ROOT NORMALIZE)
+        endif()
+    endif()
+
+    if(NOT nanobind_ROOT)
+        message(FATAL_ERROR "Failed to find nanobind package: ${nanobind_error}")
+    endif()
+
+    jrl_find_package(nanobind ${ARGN})
+
+    message(STATUS "   Nanobind CMake directory: ${nanobind_ROOT}")
+
+    # If you install nanobind with pip, it will include tsl-robin-map in <nanobind>/ext/robin_map
+    # On macOS, brew install nanobind will not include tsl-robin-map, we need to install it via: brew install robin-map
+    # Naturally, find_package(nanobind CONFIG REQUIRED) will succeed (nanobind_FOUND -> True), but the tsl-robin-map dependency will be missing, causing build errors.
+    # So let's check if the headers are available, otherwise require tsl-robin-map explicitly.
```

Comment:
In this [document](https://nanobind.readthedocs.io/en/latest/api_cmake.html#submodule-dependencies) they speak about NB_USE_SUBMODULE_DEPS variable to set if we use nanobind as a system dependency.